ndc-lambda-sdk/src/schema.ts

```diff
@@ -190,14 +200,40 @@ export class JSONValue {
   }
 }
 
+export type TypePathSegment =
+  { segmentType: "FunctionParameter", functionName: string, parameterName: string }
+  | { segmentType: "FunctionReturn", functionName: string }
+  | { segmentType: "ObjectProperty", typeName: string, propertyName: string }
+  | { segmentType: "Array" }
+  | { segmentType: "TypeParameter", typeName: string, index: number }
+  | { segmentType: "IndexSignature", typeName: string, sigIndex: number, component: "key" | "value" }
+  | { segmentType: "UnionMember", typeName: string, memberIndex: number }
+
+export function typePathToString(segments: TypePathSegment[]): string {
+  return segments.map(typePathSegmentToString).join(", ")
+}
```

Comment:
Should this be '.' style? Otherwise, show as JSON.stringify(foo.map...)

Reply:
I don't think so, because this uses the below function to form sentences for error messages, not syntax (eg "function "test" parameter "param", array type, type "MyObject" property "prop")

Reply:
Ah I see. Sure that's ok then.